docs/en/integrations/data-ingestion/data-formats/binary.md

```diff
@@ -0,0 +1,220 @@
+
+```sql
+SELECT *
+FROM file('data.binary', RowBinary, 'path String, month Date, hits UInt32')
+LIMIT 5
+
+┌─path───────────────────────────┬──────month─┬─hits─┐
+│ Bangor_City_Forest             │ 2015-07-01 │   34 │
+│ Alireza_Afzal                  │ 2017-02-01 │   24 │
+│ Akhaura-Laksam-Chittagong_Line │ 2015-09-01 │   30 │
+│ 1973_National_500              │ 2017-10-01 │   80 │
+│ Attachment                     │ 2017-09-01 │ 1356 │
+└────────────────────────────────┴────────────┴──────┘
+```
+
+Consider using [RowBinaryWithNames](/docs/en/interfaces/formats.md/#rowbinarywithnames), which also adds a header row with a columns list. [RowBinaryWithNamesAndTypes](/docs/en/interfaces/formats.md/#rowbinarywithnamesandtypes) will also add an additional header row with column types.
+
+### Importing from RowBinary files
+To load data from a RowBinary file, we can use a `FROM INFILE` clause:
+
+```sql
+INSERT INTO sometable
+FROM INFILE 'data.binary'
+FORMAT RowBinary
+```
+
+## Importing single binary value using RawBLOB
+
+Suppose we want to read an entire binary file and save it into a field in a table. This is the case when [RawBLOB format](/docs/en/interfaces/formats.md/#rawblob) can be used. This format can be directly used with a single-column table only:
+
+```sql
+CREATE TABLE images(data String) Engine = Memory
+```
+
+Let's save an image file to the `images` table:
+
+```bash
+cat image.jpg | clickhouse-client -q "INSERT INTO images FORMAT RawBLOB"
+```
+
+We can check the `data` field length which will be equal to the original file size:
+
+```sql
+SELECT length(data) FROM images
+
+┌─length(data)─┐
+│         6121 │
+└──────────────┘
+```
+
+### Exporting RawBLOB data
+
+This format can also be used to export data using an `INTO OUTFILE` clause:
+
+```sql
+SELECT * FROM images LIMIT 1
+INTO OUTFILE 'out.jpg'
+FORMAT RawBLOB
+```
+
+Note that we had to `LIMIT 1` because exporting more than a single value will create a corrupted file.
+
+## MessagePack
+
+ClickHouse supports importing and exporting to [MessagePack](https://msgpack.org/) using [MsgPack](/docs/en/interfaces/formats.md/#msgpack). To export to MessagePack format:
+
+```sql
+SELECT *
+FROM some_data
+INTO OUTFILE 'data.msgpk'
+FORMAT MsgPack
+```
+
+To import data from a [MessagePack file](assets/data.msgpk):
+
+```sql
+INSERT INTO sometable
+FROM INFILE 'data.msgpk'
+FORMAT MsgPack
+```
+
+## Protocol Buffers
+
+To work with [Protocol Buffers](/docs/en/interfaces/formats.md/#protobuf) we first need to define a [schema file](assets/schema.proto):
+
+```protobuf
+syntax = "proto3";
+
+message MessageType {
+  string path = 1;
+  date month = 2;
+  uint32 hits = 3;
+};
+```
+
+Path to this schema file (`schema.proto` in our case) is set in a `format_schema` settings potion for [Protobuf](/docs/en/interfaces/formats.md/#protobuf) format:
+
+```sql
+SELECT * FROM some_data
```

Comment:
can the user reproduce this example? where is some_data?

Reply:
Used this table all over the docs, but didn't give a DDL/dump. Do you think I should mention it somewhere and let download data for it? It looks like table structure and data isn't important in such cases, since formatting clauses are focused on.

Reply:
Maybe a intro page referencing where users can find the data and how to load it?

Reply:
or just do what you've done in a lot of cases which is explicitly point out where they can find the data for each example. good with either.